zerver/webhooks/bitbucket2/view.py

```diff
@@ -142,11 +142,12 @@ def get_type(request: HttpRequest, payload: Dict[str, Any]) -> str:
         # Note that we only need the HTTP header to determine pullrequest events.
         # We rely on the payload itself to determine the other ones.
         event_key = validate_extract_webhook_http_header(request, "X_EVENT_KEY", "BitBucket")
-        action = re.match('pullrequest:(?P<action>.*)$', event_key)
-        if action:
-            action_group = action.group('action')
-            if action_group in PULL_REQUEST_SUPPORTED_ACTIONS:
-                return pull_request_template.format(action_group)
+        if event_key is not None:
+            action = re.match('pullrequest:(?P<action>.*)$', event_key)
+            if action:
+                action_group = action.group('action')
+                if action_group in PULL_REQUEST_SUPPORTED_ACTIONS:
+                    return pull_request_template.format(action_group)
```

Comment:
I don't think this is the right fix -- it should be `assert event_key is not None` instead.  Because we are passing effectively fatal=True, so that assertion is correct.  It's generally bad practice to add useless if statements; makes the code more confusing as it suggests the alternative is a valid possibility to think about.  So assert is the better approach :)